### ISSUE TITLE
differents with model.eval() and model.train()?

### ISSUE THREAD

**Issue by wongkwan** [facebookresearch/VideoPose3D#113]
Why are model.eval() and model.train() different after the same epoch?Which result should we rely on? I'd appreciate it if you could help me out.

**Comment by dariopavllo** [comment]
The results are different because the model is trained with [batch normalization](https://pytorch.org/docs/stable/nn.html#batchnorm1d). For training, the model uses batch statistics. For testing, it uses the running statistics. You should rely on the latter when evaluating the model (i.e. `model.eval()`). This is the correct practice in all models that use batch normalization.
